src/component.js

```diff
@@ -61,16 +61,32 @@ Component.prototype.setState = function(update, callback) {
  * re-renderd
  */
 Component.prototype.forceUpdate = function(callback) {
-	let vnode = this._vnode, dom = this._vnode._dom, parentDom = this._parentDom;
+	let vnode = this._vnode, oldDom = this._vnode._dom, parentDom = this._parentDom;
 	if (parentDom) {
 		// Set render mode so that we can differantiate where the render request
 		// is coming from. We need this because forceUpdate should never call
 		// shouldComponentUpdate
 		const force = callback!==false;
 
 		let mounts = [];
-		diff(parentDom, vnode, assign({}, vnode), this._context, parentDom.ownerSVGElement!==undefined, null, mounts, force, dom == null ? getDomSibling(vnode) : dom);
+		let newDom = diff(parentDom, vnode, assign({}, vnode), this._context, parentDom.ownerSVGElement!==undefined, null, mounts, force, oldDom == null ? getDomSibling(vnode) : oldDom);
 		commitRoot(mounts, vnode);
+
+		if (newDom != oldDom) {
+			updateParentDomPointers(vnode);
+
+			// // 3496 B
```

Comment:
TODO: Remove all the commented code before checking in. Wanted to first give people an idea of some of the golfing techniques I tried.

P.S. Surprised this iterative approach isn't smaller than the recursive function:

Minified:
```js
// Iterative: 3496 B
for (; null != (o = o.__p) && null != o.__c; )
    for (o.__e = o.__c.base = null, r = 0; r < o.__k.length; r++)
        if (null != (i = o.__k[r]) && null != i.__e) {
            o.__e = o.__c.base = i.__e;
            break;
        }
```

```js
// Recursive: 3484 B
function m(n) {
	var l, u;
	if (null != (n = n.__p) && null != n.__c) {
		for (n.__e = n.__c.base = null, l = 0; l < n.__k.length; l++)
			if (null != (u = n.__k[l]) && null != u.__e) {
				n.__e = n.__c.base = u.__e;
				break;
			}
		return m(n);
	}
}
```

Reply:
Do we have some guidelines about bytes saving vs. function call overhead? Asking for a friend 😂

Reply:
I've seen a bench (assuming it can be trusted 🤷‍♂) where in one situation recursion was faster than the iterative solution, so I typically go for the byte savings first on something like this, especially since this code is only called once per update.

I think it would be hard to measure the perf difference between an iterative solution and recursive in this case since it isn't called very often in a diff (as opposed to the children loop in `diffChildren` which is called for every VNode, for example).

But that's just me. I don't spend as much time doing perf tests as @developit or others, so I defer to them.